hbase_sql/src/main/java/com/splicemachine/stream/StreamListener.java

```diff
@@ -403,6 +429,36 @@ public void failed(Throwable e) {
     public Throwable getFailure() {
         return failure;
     }
+    
+    private void manageStreaming() {
```

Comment:
Does this need synchronization? Is it possible to call `pauseStreaming` and `continueStreaming` almost at the same time and end up in a weird state? Like sending `StreamProtocol.PauseStream`, then `StreamProtocol.ContinueStream`, then setting `paused=false` and `pause=true` ?

Reply:
added synchronization on pauseStreaming and continueStreaming